chainntnfs/txconfnotifier.go

```diff
@@ -297,6 +322,45 @@ func (tcn *TxConfNotifier) ConnectTip(blockHash *chainhash.Hash,
 		}
 	}
 
+	// In order to update the height hint for all the required transactions
+	// under one database transaction, we'll gather the set of unconfirmed
+	// transactions along with the ones that confirmed at the current
+	// height. To do so, we'll iterate over the confNotifications map, which
+	// contains the transactions we currently have notifications for. Since
+	// this map doesn't tell us whether the transaction hsa confirmed or
+	// not, we'll need to look at txsByInitialHeight to determine so.
+	var txs []chainhash.Hash
+	for confirmedTx := range tcn.txsByInitialHeight[tcn.currentHeight] {
+		txs = append(txs, confirmedTx)
+	}
+out:
+	for maybeUnconfirmedTx := range tcn.confNotifications {
+		for height, confirmedTxs := range tcn.txsByInitialHeight {
+			// Skip the transactions that confirmed at the new block
+			// height as those have already been added.
+			if height == blockHeight {
+				continue
+			}
+
+			// If the transaction was found within the set of
+			// confirmed transactions at this height, we'll skip it.
+			if _, ok := confirmedTxs[maybeUnconfirmedTx]; ok {
+				continue out
+			}
+		}
+		txs = append(txs, maybeUnconfirmedTx)
+	}
+
+	if len(txs) > 0 {
+		err := tcn.hintCache.CommitConfirmHint(tcn.currentHeight, txs...)
```

Comment:
Could use a `Tracef` log here and below when updating height hints.

Reply:
Decided to keep the log statements within the calls themselves to avoid having to do it everywhere.